ngraph/core/reference/src/runtime/reference/tile.cpp

```diff
@@ -22,26 +22,87 @@
 
 using namespace ngraph;
 
-void runtime::reference::tile(
-    const char* arg, char* out, const Shape& in_shape, const Shape& out_shape, size_t elem_size)
+bool Increase(const Shape& shape, std::vector<int64_t>& indices, size_t& axis, bool& run)
+{
+    if (axis-- == 0)
+    {
+        run = false;
+        return false;
+    }
+
+    if (++indices[axis] != shape[axis])
+    {
+        axis = indices.size();
+        return false;
+    }
+
+    indices[axis] = 0;
+    return true;
+}
+
+std::vector<int64_t> createPitches(const Shape& dims)
+{
+    std::vector<int64_t> pitch;
+    auto tensor_rank = dims.size();
+
+    for (int i = 0; i < tensor_rank - 1; i++)
+    {
+        int64_t val =
+            std::accumulate(dims.begin() + i + 1, dims.end(), 1, std::multiplies<int64_t>());
```

Comment:
std::inclusive_scan is a C++17 feature, so we cannot use it right now, please use what's available in C++11.

@ilyachur do we have a plan to migrate to newer C++ standard versions? Even if so, probably migration will take time, so maybe it's worth considering as an intermediate step to introduce a library which implements some of C++17 features in C++11  (something like [abseil](https://abseil.io/)) ?

Reply:
Hi at the current moment we cannot (and don't have a plan) migrate to newer C++. Because OpenVINO should support default compiler on supported platforms. Not for all platforms default compilers support C++17.